source/gems/dry-validation/input-preprocessing.html.md

```diff
@@ -0,0 +1,81 @@
+---
+title: Input Preprocessing
+layout: gem-single
+---
+
+Sometimes, the data coming from outside needs to be preprocessed before being validated. Think of extraneous spaces before or after a string, for example.
+
+### Preprocessing names to remove spaces
+
+In the context of a web application form, it is common to ask for people's names. The name should not contain spaces before or after the other characters. It would be possible to reject such strings as invalid, but that would make the application harder to use. Instead, we can let dry-validation preprocess the input to remove such spaces. To that end, we will create a new type of validator:
+
+```ruby
+require "dry-validation"
+
+module Types
+  include Dry::Types.module
+
+  Name = Types::String.constructor do |str|
+    str ? str.strip.chomp : str
+  end
+end
+
+SignUpForm = Dry::Validation.Form do
+  configure do
+    config.type_specs = true
+  end
+
+  required(:email, :string).filled(format?: /.@.+[.][a-z]{2,}/i)
+  required(:name, Types::Name).filled(min_size?: 1)
+  required(:password, :string).filled(min_size?: 6)
+end
+
+result = SignUpForm.call(
+  "name" => "\t François \n",
+  "email" => "francois@teksol.info",
+  "password" => "some password")
+
+result.success?
+# true
+
+result.output.fetch(:name)
```

Comment:
Actually, this could just be `result[:name]` as `Result` objects are hash-like. Also, `output` is going to be deprecated prior 1.0.0 in favor of `to_h`